### PR TITLE
th/platform_route_pt2

### PR DESCRIPTION
**Pull request by thom311**
Adjust caching of routes in platform.

(the last commit is still WIP. Please review the rest :) tks)

### CODE REVIEW

File: shared/nm-utils/nm-dedup-multi.h [hunk 1]
```diff
@@ -219,6 +219,18 @@ struct _NMDedupMultiHeadEntry {
 	CList lst_idx;
 };
 
+/*****************************************************************************/
+
+static inline gconstpointer
+nm_dedup_multi_entry_get_obj (const NMDedupMultiEntry *entry)
+{
+	/* convenience method that allows to skip the %NULL check on
+	 * @entry. Think of the NULL-conditinal operator ?. of C# */
```
Comment (fgiudici) [inline shared/nm-utils/nm-dedup-multi.h]: conditiOnal

File: shared/nm-utils/nm-dedup-multi.c [hunk 2]
```diff
@@ -994,6 +994,64 @@ nm_dedup_multi_objs_to_ptr_array_head (const NMDedupMultiHeadEntry *head_entry,
...
+ * nm_dedup_multi_entry_reorder:
+ * @entry: the entry to reorder. It must not be NULL (and tracked in an index).
+ * @entry_order: (allow-none): an optional other entry. It MUST be in the same
+ *   list as entry. If given, @entry will be ordered after/before @entry_order.
+ *   If left at %NULL, @entry will be moved to the front/end of the list.
+ * @order_after: if @entry_order is given then %TRUE means to order @entry after
+ *   @entry_order (otherwise before). If @entry_order is %NULL, %TRUE means to
+ *   move @entry to the end of the list (otherwise the beginning).
+ *   Note that @order_after %TRUE, means to link @entry after @entry_order,
+ *   but it means to link @entry before the list's head (to become the tail).
```
Comment (fgiudici) [inline shared/nm-utils/nm-dedup-multi.c]: Well, not very clear to me what this last sentence mean... if @order_after is TRUE, then @entry will never be linked before the list's HEAD, isn't it?

Comment (thom311) [inline shared/nm-utils/nm-dedup-multi.c]: CList is a circular list. So, linking "before" the head, means to link at the tail of the list. order_after=TRUE means to link at the tail -- hence: before the head.

Anyway, I reword the comment...

Comment (fgiudici) [inline shared/nm-utils/nm-dedup-multi.c]: Thanks, now I got it

**Comment by thom311** [comment]
Thanks. Addressed comments and repushed branch

**Comment by thom311** [comment]
merged as c0ac4a2fdd530998bb6fa28f85d6f80467295550

**Pull request closed without merging**